omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/RegisterPatientFragmentController.java

```diff
@@ -181,6 +182,13 @@ public FragmentActionResult submit(UiSessionContext sessionContext, @RequestPara
             registrationData.addBiometricData(new BiometricData(subject, identifierType));
         }
 
+        if (StringUtils.isNotBlank(fingerprintSubjectId)) {
+            BiometricSubject biometricSubject = new BiometricSubject();
+            biometricSubject.setSubjectId(fingerprintSubjectId);
+            BiometricData biometricData = registrationService.generateBiometricData(biometricSubject);
+            registrationData.getBiometrics().add(biometricData);
```

Comment:
Yes, we need this because of the structure of the biometrics interfaces. I will move the generateBiometricData method to this module and I will mark it as private.